src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java

```diff
@@ -43,9 +45,17 @@
     public DiscountResponseVO findById(Long id) {
         Discount discount = discountRepository.findById(id)
                 .orElseThrow(() -> new EntityNotFoundException(Discount.class, "id", id));
-        Long quantity = discount.getViewNumber() == null ? 1 : discount.getViewNumber() + 1;
-        discount.setViewNumber(quantity);
-        discountRepository.save(discount);
+        while (true) {
+            try {
```

Comment:
Before retrying it is preferable to read the entity again in a new transaction one more time. Will discuss on our daily call